nmrfx-structure/src/main/java/org/nmrfx/structure/chemistry/energy/Dihedral.java

```diff
@@ -425,17 +425,18 @@ public void setBoundaries(final double sigma, boolean useDegrees, double maxRang
             inputSigma[i] = sigma;
         }
         int aStart = nPseudoAngles * 2 * stepSize;
+        List<String> parentNamesToScale = Arrays.asList("P", "O5'", "C5'", "C4'", "O3'", "CA", "C");
         for (int iAtom = 0; iAtom < angleAtoms.size(); iAtom++) {
             Atom atom = angleAtoms.get(iAtom).daughterAtom;
             // temporarily set everything to use sigma, till we make sure counting is correct
             if (atom == null) {
-                throw new NullPointerException("daughter null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
+                throw new IllegalStateException("daughter null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
             } else if (atom.parent == null) {
-                throw new NullPointerException("parent null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
+                throw new IllegalStateException("parent null " + angleAtoms.get(iAtom).daughterAtom.getFullName());
```

Comment:
I know that this is exactly what I suggested, but if we decide to use requireNonNull() in some places, it may be a good idea to generalize it for all null checks? The main differences I see are the type of the thrown exception, and the time at which the name is created. 

Here the call to angleAtoms.get(..) is only done if atom is null, and it would be done all the time using requireNonNull(), so this version is slightly better performance-wise while the other is more readable. I'm honestly not sure which one we should prefer.